internal/impl/timeplus/output.go

```diff
@@ -0,0 +1,216 @@
+` + "```" + `
+
+A sample config to send the data to Timeplus Enterprise Cloud
+` + "```yml" + `
+output:
+  timeplus:
+    workspace: my_workspace_id
+    stream: mystream
+    apikey: <Api key genereated on web console>
+` + "```" + `
+
+A sample config to send the data to Timeplus Enterprise (self-hosted)
+` + "```yml" + `
+output:
+  timeplus:
+    url: http://localhost:8000
+    workspace: my_workspace_id
+    stream: mystream
+    username: username
+    password: pw
+` + "```" + `
+
+A sample config to send the data to timeplusd
+` + "```yml" + `
+output:
+  timeplus:
+    url: http://localhost:3218
+    stream: mystream
+    username: username
+    password: pw
+` + "```" + `
+`)
+
+	outputConfigSpec.
+		Field(service.NewStringEnumField("target", http.TargetTimeplus, http.TargetTimeplusd).Default(http.TargetTimeplus).Description("The destination type, either Timeplus Enterprise or timeplusd")).
+		Field(service.NewURLField("url").Description("The url should always include schema and host.").Default("https://us-west-2.timeplus.cloud").Examples("http://localhost:8000", "http://127.0.0.1:3218")).
+		Field(service.NewStringField("workspace").Optional().Description("ID of the workspace. Required if target is `timeplus`.")).
+		Field(service.NewStringField("stream").Description("The name of the stream. Make sure the schema of the stream matches the input")).
+		Field(service.NewStringField("apikey").Secret().Optional().Description("The API key. Required if you are sending message to Timeplus Enterprise Cloud")).
+		Field(service.NewStringField("username").Optional().Description("The username. Required if you are sending message to Timeplus Enterprise (self-hosted) or timeplusd")).
+		Field(service.NewStringField("password").Secret().Optional().Description("The password. Required if you are sending message to Timeplus Enterprise (self-hosted) or timeplusd")).
+		Field(service.NewOutputMaxInFlightField()).
+		Field(service.NewBatchPolicyField("batching"))
+
+	if err := service.RegisterBatchOutput("timeplus", outputConfigSpec, newTimeplusOutput); err != nil {
+		panic(err)
+	}
+}
+
+type timeplus struct {
+	logger *service.Logger
+	client Writer
+}
+
+// Close implements service.Output
+func (t *timeplus) Close(context.Context) error {
+	// TODO: shall we wait for ongoing writes?
+	t.client = nil
+	return nil
+}
+
+// Connect implements service.Output
+func (t *timeplus) Connect(context.Context) error {
+	if t.client == nil {
+		return errors.New("client not initialized")
+	}
+
+	return nil
+}
+
+func (t *timeplus) WriteBatch(ctx context.Context, b service.MessageBatch) error {
+	if len(b) == 0 {
+		return nil
+	}
+
+	cols := []string{}
+	rows := [][]any{}
+
+	// Here we assume all messages have the same structure, same keys
+	// Currently we will just skip the message if it is invalid. Probably we should add a feature flag to not skip them by propagating the error
```

Comment:
Normally, Connect outputs return an error if a message fails. This enables users to leverage a `fallback` output so they can implement a dead letter queue for example. They can use a `drop` output in there if they genuinely want to drop failed messages.

Reply:
Makes sense. Updated. Now it will just propagate the error.